src/fmt/writer/termcolor/extern_impl.rs

```diff
@@ -70,46 +71,72 @@ impl Formatter {
 
 pub(in crate::fmt::writer) struct BufferWriter {
     inner: termcolor::BufferWriter,
-    test_target: Option<Target>,
+    test_target_type: Option<TargetType>,
+    target_pipe: Option<Arc<Mutex<dyn io::Write + Send + 'static>>>,
```

Comment:
Can you explain why `Arc<Mutex<>>` is needed?

Reply:
Because the pipe is still partly owned by the `Builder` (it doesn't get deconstructed on `build()`).